fenix/app/src/main/java/org/mozilla/gecko/search/SearchWidgetProvider.kt

```diff
@@ -163,19 +168,19 @@ class SearchWidgetProvider : AppWidgetProvider() {
         }
     }
 
-    private fun RemoteViews.setIcon(context: Context) {
+    private fun RemoteViews.setIcon(context: Context, viewId: Int, resourceId: Int) {
```

Comment:
This change has some cons to it. Line 188 and below always sets the same content description (giving `R.id.button_search_widget_voice` the wrong description). Also, a nit, microphone icon doesn't have gradient so there is no value to run it through the same version check.

If we can't sen the mic icon through xml, maybe changing the function to `setIcons` and just adding the icon and content here would make more sense? 

It looks like `setIcon` method was supposed to encapsulate icon logic. Making a `setIcons` method or two separate methods would help to keep the icon related code in one place, and make the `createRemoteViews` more straightforward.

Reply:
I believe this doesn't change the content description for `button_search_widget_voice`, only for `button_search_widget_new_tab_icon`, however I agree we should try to use the xml to change the icon.